app/js/components/tabs/tabcomponents/programmeComponent.jsx

```diff
@@ -22,63 +38,233 @@ class ProgrammeComponent extends Component {
+                parameters.bornDuringPeriod.push({
+                    name: 'startDate', dataType: 'date', value: this.state.minAge
+                });
+            }
+            if (this.state.maxAge) {
+                parameters.bornDuringPeriod.push({
+                    name: 'endDate', dataType: 'date', value: this.state.maxAge
+                });
+            }
+        }
+        if (this.state.program) {
+            parameters.programEnrollment.push({
+                name: 'programs',
+                type: 'program',
+                value: [this.state.program]
+            });
+        }
+         if (this.state.enrolledStartDate) {
+            parameters.programEnrollment.push({
+                name: 'enrolledStartDate',
+                type: 'date',
+                value: this.state.enrolledStartDate
+            });
+        }
+        if (this.state.enrolledEndDate) {
+            parameters.programEnrollment.push({
+                name: 'enrolledEndDate',
+                type: 'date',
+                value: this.state.enrolledEndDate
+            });
+        }
+        if (this.state.completedStartDate) {
+            parameters.programEnrollment.push({
+                name: 'completedStartDate',
+                type: 'date',
+                value: this.state.completedStartDate
+            });
+        }
+        if (this.state.completedEndDate) {
+            parameters.programEnrollment.push({
+                name: 'completedEndDate',
+                type: 'date',
+                value: this.state.completedEndDate
+            });
+        }
+        return parameters;
+    }
+
     // Get program workflows for the program selected
-    getWorkflow() {
-        const program = document.getElementById("select-program").value;
-        this.props.fetchData(`/program/${program}`).then(data => {
-            this.setState({
-                workflows: data.allWorkflows
+    getWorkflow(program) {
+        if (program) {
+            this.props.fetchData(`/program/${program}`).then(data => {
+                this.setState({
+                    workflows: data.allWorkflows
+                });
+                this.getStates(data.allWorkflows);
             });
-            this.getStates(data.allWorkflows);
-        });
+        } else {
+            this.setState({ workflows: [], workflow: '' });
+        }
     }
     // Get the states from the workflow.
     getStates(workflows) {
         /** Update state based on the data retreived from the workflows
          * @TODO: State data does not exist on the local, but exists on refapp,
          * try to populate states on the local
         */
+        // console.log('work flows - ', workflows)
         
     }
     render() {
         let programs = this.state.programs.map((program) => {
             return (
-                <option key={program.uuid} value={program.uuid}>
+                <option key={program.uuid} value={program.name}>
```

Comment:
Yes, you are right. I was using the value to get the descriptive name for the label and also to make a search request.
I will refactor this to make the search request by the uuid, and the descriptive name for the label.